--- configure.in.old	2011-05-01 14:22:44.000000000 +0100
+++ configure.in	2011-05-01 14:23:11.000000000 +0100
@@ -76,7 +76,7 @@
 
 if test x$os = xdarwin; then
     AC_MSG_NOTICE([Using vDSP on OS X])
-    LIBS="$LIBS -framework Accelerate"
+    LIBS="$LIBS -XCClinker -framework -XCClinker Accelerate"
     FFT_WRAPPER="fftlibvdsp_op.cpp"
     AM_CONDITIONAL(FFTW3, false)
     AM_CONDITIONAL(VDSP, true)
