--- lib/fftlibvdsp_op.cpp.orig	2011-11-03 19:28:10.000000000 +0000
+++ lib/fftlibvdsp_op.cpp	2011-11-03 19:29:01.000000000 +0000
@@ -23,12 +23,12 @@
 	{
 		delete[] A.realp;
 		delete[] A.imagp;
-		destroy_fftsetupD(SetupReal);
+		vDSP_destroy_fftsetupD(SetupReal);
 	}
 		
 	A.realp = new double[ N/2];
 	A.imagp = new double[ N/2];
-	SetupReal = create_fftsetupD(Exp, 0);
+	SetupReal = vDSP_create_fftsetupD(Exp, 0);
 	Init = true;
 }
 
@@ -46,9 +46,9 @@
 void 
 FFTLib_op::ComputeFrame(int N, double *in, double *out)
 {
-	ctozD ((DSPDoubleComplex*) in, 2, &A, 1, N/2 );
+	vDSP_ctozD ((DSPDoubleComplex*) in, 2, &A, 1, N/2 );
 	
-	fft_zripD(SetupReal, &A, 1, Exp, FFT_FORWARD);
+	vDSP_fft_zripD(SetupReal, &A, 1, Exp, FFT_FORWARD);
 	
 	int i,j;
 	for (i=0; i<N/2; i++)
