Index: libproxy/CMakeLists.txt
===================================================================
--- libproxy/CMakeLists.txt	(revision 823)
+++ libproxy/CMakeLists.txt	(working copy)
@@ -6,7 +6,7 @@
 if (WIN32)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS=1)
 else(WIN32)
-  add_definitions(-D_POSIX_C_SOURCE=1)
+  add_definitions(-D_POSIX_C_SOURCE=200112L)
   set(CMAKE_CXX_FLAGS "-fvisibility=hidden ${CMAKE_CXX_FLAGS}")
 endif(WIN32)
 
