--- ../old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/configure	2010-07-15 19:31:59.000000000 +0000
+++ gst-libs/ext/ffmpeg/configure	2010-08-17 09:18:43.000000000 +0000
@@ -2729,7 +2729,6 @@
 check_cflags -Wwrite-strings
 check_cflags -Wtype-limits
 check_cflags -Wundef
-check_cflags -Wmissing-prototypes
 enabled extra_warnings && check_cflags -Winline
 
 # add some linker flags
@@ -2788,7 +2787,6 @@
 elif enabled gcc; then
     check_cflags -fno-tree-vectorize
     check_cflags -Werror=implicit
-    check_cflags -Werror=missing-prototypes
 elif enabled clang; then
     check_cflags -Qunused-arguments
 elif enabled armcc; then
