--- sys/directsound/gstdirectsoundsink.c.orig	2015-07-21 16:54:53.405817179 +0000
+++ sys/directsound/gstdirectsoundsink.c	2015-07-21 16:55:02.145818763 +0000
@@ -399,10 +399,10 @@
 {
   GstDirectSoundSink *dsoundsink;
   HRESULT hRes;
+  GUID* device = NULL;
 
   dsoundsink = GST_DIRECTSOUND_SINK (asink);
 
-  GUID* device = NULL;
   if (dsoundsink->has_device) {
     device = &dsoundsink->device;
   }
