--- gst-libs/gst/video/video-dither.c.orig	2015-07-21 16:15:26.773673474 +0000
+++ gst-libs/gst/video/video-dither.c	2015-07-21 16:15:32.493672873 +0000
@@ -413,7 +413,6 @@
   }
   dither->orc_mask64 = GUINT64_FROM_BE (dither->orc_mask64);
   dither->orc_mask32 = GUINT32_FROM_BE (dither->orc_mask32);
-  GST_DEBUG ("mask64 %08llx", (unsigned long long int) dither->orc_mask64);
   GST_DEBUG ("mask32 %08x", dither->orc_mask32);
 
   switch (method) {
