diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/cscd.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/cscd.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/cscd.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/cscd.c	2010-08-17 10:54:53.000000000 +0000
@@ -20,6 +20,7 @@
  */
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/types.h>
 
 #include "avcodec.h"
 
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/dxa.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/dxa.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/dxa.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/dxa.c	2010-08-17 10:55:17.000000000 +0000
@@ -30,6 +30,7 @@
 #include "libavutil/intreadwrite.h"
 #include "avcodec.h"
 
+#include <sys/types.h>
 #include <zlib.h>
 
 /*
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/flashsv.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/flashsv.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/flashsv.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/flashsv.c	2010-08-17 10:55:34.000000000 +0000
@@ -53,6 +53,7 @@
 #include "avcodec.h"
 #include "get_bits.h"
 
+#include <sys/types.h>
 #include <zlib.h>
 
 typedef struct FlashSVContext {
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/flashsvenc.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/flashsvenc.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/flashsvenc.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/flashsvenc.c	2010-08-17 10:55:45.000000000 +0000
@@ -56,6 +56,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/types.h>
 #include <zlib.h>
 
 #include "avcodec.h"
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/lcldec.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/lcldec.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/lcldec.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/lcldec.c	2010-08-17 10:56:06.000000000 +0000
@@ -40,6 +40,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/types.h>
 
 #include "avcodec.h"
 #include "bytestream.h"
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/lclenc.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/lclenc.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/lclenc.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/lclenc.c	2010-08-17 10:56:26.000000000 +0000
@@ -40,6 +40,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <sys/types.h>
 
 #include "avcodec.h"
 #include "lcl.h"
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/pngdec.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/pngdec.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/pngdec.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/pngdec.c	2010-08-17 10:56:41.000000000 +0000
@@ -27,6 +27,7 @@
  * - add 2, 4 and 16 bit depth support
  */
 
+#include <sys/types.h>
 #include <zlib.h>
 
 //#define DEBUG
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/pngenc.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/pngenc.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/pngenc.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/pngenc.c	2010-08-17 10:56:49.000000000 +0000
@@ -27,6 +27,7 @@
  * - add 2, 4 and 16 bit depth support
  */
 
+#include <sys/types.h>
 #include <zlib.h>
 
 //#define DEBUG
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/svq3.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/svq3.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/svq3.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/svq3.c	2010-08-17 10:57:37.000000000 +0000
@@ -53,6 +53,7 @@
 #include "vdpau_internal.h"
 
 #if CONFIG_ZLIB
+#include <sys/types.h>
 #include <zlib.h>
 #endif
 
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tiff.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tiff.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tiff.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tiff.c	2010-08-17 10:57:47.000000000 +0000
@@ -26,6 +26,7 @@
  */
 #include "avcodec.h"
 #if CONFIG_ZLIB
+#include <sys/types.h>
 #include <zlib.h>
 #endif
 #include "lzw.h"
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tiffenc.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tiffenc.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tiffenc.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tiffenc.c	2010-08-17 10:57:54.000000000 +0000
@@ -26,6 +26,7 @@
  */
 #include "avcodec.h"
 #if CONFIG_ZLIB
+#include <sys/types.h>
 #include <zlib.h>
 #endif
 #include "bytestream.h"
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tscc.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tscc.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tscc.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/tscc.c	2010-08-17 10:58:04.000000000 +0000
@@ -41,6 +41,7 @@
 #include "avcodec.h"
 #include "msrledec.h"
 
+#include <sys/types.h>
 #include <zlib.h>
 
 
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/zmbv.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/zmbv.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/zmbv.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/zmbv.c	2010-08-17 10:58:20.000000000 +0000
@@ -30,6 +30,7 @@
 #include "libavutil/intreadwrite.h"
 #include "avcodec.h"
 
+#include <sys/types.h>
 #include <zlib.h>
 
 #define ZMBV_KEYFRAME 1
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/zmbvenc.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/zmbvenc.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/zmbvenc.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavcodec/zmbvenc.c	2010-08-17 10:58:30.000000000 +0000
@@ -30,6 +30,7 @@
 #include "libavutil/intreadwrite.h"
 #include "avcodec.h"
 
+#include <sys/types.h>
 #include <zlib.h>
 
 #define ZMBV_KEYFRAME 1
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavformat/matroskadec.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavformat/matroskadec.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavformat/matroskadec.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavformat/matroskadec.c	2010-08-17 10:53:53.000000000 +0000
@@ -41,6 +41,7 @@
 #include "libavutil/intreadwrite.h"
 #include "libavutil/avstring.h"
 #include "libavutil/lzo.h"
+#include <sys/types.h>
 #if CONFIG_ZLIB
 #include <zlib.h>
 #endif
diff -ur -x configure -x config.mak -x Makefile.in -x Makefile.am -x configure.ac old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavformat/mov.c gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavformat/mov.c
--- old/gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavformat/mov.c	2010-07-15 19:31:59.000000000 +0000
+++ gst-ffmpeg-0.10.11/gst-libs/ext/ffmpeg/libavformat/mov.c	2010-08-17 10:54:10.000000000 +0000
@@ -34,6 +34,7 @@
 #include "libavcodec/mpeg4audio.h"
 #include "libavcodec/mpegaudiodata.h"
 #include "libavcodec/get_bits.h"
+#include <sys/types.h>
 
 #if CONFIG_ZLIB
 #include <zlib.h>
