--- gst-plugins-good-0.10.24/sys/directsound/gstdirectsoundsink.c	2014-03-29 12:45:34.734215372 +0000
+++ gst-plugins-good-0.10.24-new/sys/directsound/gstdirectsoundsink.c	2014-03-29 12:44:19.210218226 +0000
@@ -102,7 +102,8 @@
 
 enum {
   ARG_0,
-  ARG_VOLUME
+  ARG_VOLUME,
+  ARG_DEVICE
 };
 
 GST_BOILERPLATE (GstDirectSoundSink, gst_directsound_sink, GstBaseAudioSink,
@@ -236,6 +237,9 @@
   g_object_class_install_property (gobject_class, ARG_VOLUME,
       g_param_spec_double ("volume", "Volume", "Volume of this stream",
           0, 1.0, 1.0, G_PARAM_READWRITE));
+  g_object_class_install_property (gobject_class, ARG_DEVICE,
+      g_param_spec_pointer ("device", "Device", "Pointer to a GUID",
+          G_PARAM_WRITABLE));
 
   gstbasesink_class->event = GST_DEBUG_FUNCPTR (gst_directsound_sink_event);
 
@@ -249,6 +253,7 @@
 {
   dsoundsink->dsoundbuffer = NULL;
   dsoundsink->volume = 1.0;
+  dsoundsink->has_device = 0;
 }
 
 static gboolean
@@ -331,12 +336,22 @@
     guint prop_id, const GValue *value , GParamSpec *pspec)
 {
   GstDirectSoundSink *sink = GST_DIRECTSOUND_SINK (object);
+  void* ptr = NULL;
 
   switch (prop_id) {
     case ARG_VOLUME:
       sink->volume = g_value_get_double (value);
       gst_directsound_sink_set_volume (sink);
       break;
+    case ARG_DEVICE:
+      ptr = g_value_get_pointer (value);
+      if (ptr) {
+        sink->has_device = 1;
+        memcpy(&sink->device, ptr, sizeof(GUID));
+      } else {
+        sink->has_device = 0;
+      }
+      break;
     default:
       G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
       break;
@@ -477,7 +492,12 @@
 
   GST_DEBUG ("Opening DirectSound Device");
 
-  if (FAILED (hr = DirectSoundCreate8 (NULL, &dsoundbuffer->pDS8, NULL))) {
+  GUID* device = NULL;
+  if (dsoundbuffer->dsoundsink->has_device) {
+    device = &dsoundbuffer->dsoundsink->device;
+  }
+
+  if (FAILED (hr = DirectSoundCreate8 (device, &dsoundbuffer->pDS8, NULL))) {
     GST_ELEMENT_ERROR (dsoundbuffer->dsoundsink, RESOURCE, FAILED, 
       ("%S.", DXGetErrorDescription9(hr)), 
       ("Failed to create directsound device. (%X)", hr));
--- gst-plugins-good-0.10.24/sys/directsound/gstdirectsoundsink.h	2014-03-29 12:45:34.734215372 +0000
+++ gst-plugins-good-0.10.24-new/sys/directsound/gstdirectsoundsink.h	2014-03-29 12:44:19.210218226 +0000
@@ -83,6 +83,9 @@
 
   /* current volume */
   gdouble volume;
+
+  int has_device;
+  GUID device;
 };
 
 struct _GstDirectSoundSinkClass
