--- gst/librfb/Makefile.in-old	2010-07-25 22:30:22.118690071 +0100
+++ gst/librfb/Makefile.in	2010-07-25 22:30:32.476425300 +0100
@@ -525,7 +525,7 @@
 plugin_LTLIBRARIES = libgstrfbsrc.la
 libgstrfbsrc_la_SOURCES = gstrfbsrc.c
 libgstrfbsrc_la_CFLAGS = $(GST_PLUGINS_BASE_CFLAGS) $(GST_CFLAGS) $(X11_CFLAGS) -I$(srcdir)/..
-libgstrfbsrc_la_LIBADD = $(GST_BASE_LIBS) $(X11_LIBS) librfb.la
+libgstrfbsrc_la_LIBADD = $(GST_BASE_LIBS) $(X11_LIBS) librfb.la -lws2_32
 libgstrfbsrc_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
 libgstrfbsrc_la_LIBTOOLFLAGS = --tag=disable-static
 librfb_la_SOURCES = \
