diff -ur -x configure -x config.mak -x Makefile.in -x '*.c' old/gst-ffmpeg-0.10.11/configure.ac gst-ffmpeg-0.10.11/configure.ac
--- old/gst-ffmpeg-0.10.11/configure.ac	2010-07-15 19:15:23.000000000 +0000
+++ gst-ffmpeg-0.10.11/configure.ac	2010-08-18 13:23:54.000000000 +0000
@@ -272,6 +272,10 @@
   FFMPEG_LIBS="\$(top_builddir)/gst-libs/ext/ffmpeg/libavformat/libavformat.a \
                \$(top_builddir)/gst-libs/ext/ffmpeg/libavcodec/libavcodec.a \
                \$(top_builddir)/gst-libs/ext/ffmpeg/libavutil/libavutil.a"
+  FFMPEG_LIBS_L="-L\$(top_builddir)/gst-libs/ext/ffmpeg/libavformat -lavformat \
+                 -L\$(top_builddir)/gst-libs/ext/ffmpeg/libavcodec -lavcodec \
+                 -L\$(top_builddir)/gst-libs/ext/ffmpeg/libavutil -lavutil"
+
   dnl
   POSTPROC_CFLAGS="-I \$(top_srcdir)/gst-libs/ext/ffmpeg/libpostproc	\
                    -I \$(top_srcdir)/gst-libs/ext/ffmpeg/libavutil 	\
@@ -283,6 +287,8 @@
   dnl libgstpostproc.la: libs to statically link to
   POSTPROC_LIBS="\$(top_builddir)/gst-libs/ext/ffmpeg/libpostproc/libpostproc.a \
   		 \$(top_builddir)/gst-libs/ext/ffmpeg/libavutil/libavutil.a"
+  POSTPROC_LIBS_L="-L\$(top_builddir)/gst-libs/ext/ffmpeg/libpostproc -lpostproc \
+  		   -L\$(top_builddir)/gst-libs/ext/ffmpeg/libavutil -lavutil"
   				 
   dnl
   SWSCALE_CFLAGS="-I \$(top_srcdir)/gst-libs/ext/ffmpeg/libswscale	\
@@ -294,6 +300,8 @@
   dnl libgstswscale.la: libs to statically link to
   SWSCALE_LIBS="\$(top_builddir)/gst-libs/ext/ffmpeg/libswscale/libswscale.a \
                 \$(top_builddir)/gst-libs/ext/ffmpeg/libavutil/libavutil.a"
+  SWSCALE_LIBS_L="-L\$(top_builddir)/gst-libs/ext/ffmpeg/libswscale -lswscale \
+                  -L\$(top_builddir)/gst-libs/ext/ffmpeg/libavutil -lavutil"
 
   FFMPEG_SUBDIRS=gst-libs
   AC_DEFINE(HAVE_AVI_H)
@@ -382,11 +390,14 @@
 
 AC_SUBST(FFMPEG_CFLAGS)
 AC_SUBST(FFMPEG_LIBS)
+AC_SUBST(FFMPEG_LIBS_L)
 AC_SUBST(FFMPEG_SUBDIRS)
 AC_SUBST(POSTPROC_CFLAGS)
 AC_SUBST(POSTPROC_LIBS)
+AC_SUBST(POSTPROC_LIBS_L)
 AC_SUBST(SWSCALE_CFLAGS)
 AC_SUBST(SWSCALE_LIBS)
+AC_SUBST(SWSCALE_LIBS_L)
 AC_SUBST(WIN32_LIBS)
  
 if test x$HAVE_FFMPEG_UNINSTALLED = x1; then
diff -ur -x configure -x config.mak -x Makefile.in -x '*.c' old/gst-ffmpeg-0.10.11/ext/ffmpeg/Makefile.am gst-ffmpeg-0.10.11/ext/ffmpeg/Makefile.am
--- old/gst-ffmpeg-0.10.11/ext/ffmpeg/Makefile.am	2010-07-06 06:49:52.000000000 +0000
+++ gst-ffmpeg-0.10.11/ext/ffmpeg/Makefile.am	2010-08-18 13:18:35.000000000 +0000
@@ -21,7 +21,7 @@
 # 			  gstffmpegscale.c
 
 libgstffmpeg_la_CFLAGS = $(FFMPEG_CFLAGS) $(GST_PLUGINS_BASE_CFLAGS) $(GST_CFLAGS)
-libgstffmpeg_la_LIBADD = $(FFMPEG_LIBS) $(GST_BASE_LIBS) $(GST_PLUGINS_BASE_LIBS) -lgstaudio-$(GST_MAJORMINOR) $(LIBM) $(WIN32_LIBS) -lz $(BZ2_LIBS)
+libgstffmpeg_la_LIBADD = $(FFMPEG_LIBS_L) $(GST_BASE_LIBS) $(GST_PLUGINS_BASE_LIBS) -lgstaudio-$(GST_MAJORMINOR) $(LIBM) $(WIN32_LIBS) -lz $(BZ2_LIBS)
 libgstffmpeg_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS) $(DARWIN_LDFLAGS)
 libgstffmpeg_la_LIBTOOLFLAGS = --tag=disable-static
 
diff -ur -x configure -x config.mak -x Makefile.in -x '*.c' old/gst-ffmpeg-0.10.11/ext/libpostproc/Makefile.am gst-ffmpeg-0.10.11/ext/libpostproc/Makefile.am
--- old/gst-ffmpeg-0.10.11/ext/libpostproc/Makefile.am	2010-07-06 06:49:52.000000000 +0000
+++ gst-ffmpeg-0.10.11/ext/libpostproc/Makefile.am	2010-08-18 13:24:38.000000000 +0000
@@ -4,7 +4,7 @@
 
 libgstpostproc_la_CFLAGS = $(POSTPROC_CFLAGS) $(GST_PLUGINS_BASE_CFLAGS) $(GST_CFLAGS) $(ORC_CFLAGS)
 libgstpostproc_la_LIBADD = -lgstvideo-@GST_MAJORMINOR@ \
-	$(POSTPROC_LIBS) \
+	$(POSTPROC_LIBS_L) \
 	$(GST_PLUGINS_BASE_LIBS) $(GST_BASE_LIBS) $(ORC_LIBS)
 
 libgstpostproc_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS) $(DARWIN_LDFLAGS)
diff -ur -x configure -x config.mak -x Makefile.in -x '*.c' old/gst-ffmpeg-0.10.11/ext/libswscale/Makefile.am gst-ffmpeg-0.10.11/ext/libswscale/Makefile.am
--- old/gst-ffmpeg-0.10.11/ext/libswscale/Makefile.am	2010-07-06 06:49:52.000000000 +0000
+++ gst-ffmpeg-0.10.11/ext/libswscale/Makefile.am	2010-08-18 13:24:58.000000000 +0000
@@ -4,7 +4,7 @@
 
 libgstffmpegscale_la_CFLAGS = $(SWSCALE_CFLAGS) $(GST_PLUGINS_BASE_CFLAGS) $(GST_BASE_CFLAGS) $(GST_CFLAGS) \
 	$(ORC_CFLAGS)
-libgstffmpegscale_la_LIBADD = $(SWSCALE_LIBS) \
+libgstffmpegscale_la_LIBADD = $(SWSCALE_LIBS_L) \
 	$(GST_PLUGINS_BASE_LIBS) -lgstvideo-$(GST_MAJORMINOR) \
 	$(GST_BASE_LIBS) $(ORC_LIBS) $(LIBM) -lz
 libgstffmpegscale_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS) $(DARWIN_LDFLAGS)
