--- src/notification_proxy.c-old	2010-07-26 01:26:54.583236947 +0100
+++ src/notification_proxy.c	2010-07-26 01:27:24.823216954 +0100
@@ -28,6 +28,8 @@
 #include "property_list_service.h"
 #include "debug.h"
 
+#include <windows.h>
+
 struct np_thread {
 	np_client_t client;
 	np_notify_cb_t cbfunc;
@@ -336,7 +338,7 @@
 			free(notification);
 			notification = NULL;
 		}
-		sleep(1);
+		Sleep(1);
 	}
 	if (npt) {
 		free(npt);
