--- configure.in.old	2011-03-12 13:21:18.161909255 +0000
+++ configure.in	2011-03-12 13:21:28.561290261 +0000
@@ -108,10 +108,5 @@
 include/musicbrainz/Makefile 
 lib/Makefile 
 osdep/Makefile
-examples/Makefile
-perl/Makefile
-python/Makefile
-python/examples/Makefile
-python/test/Makefile
 libmusicbrainz.pc], 
 echo timestamp > stamp-h)
--- Makefile.in.old	2011-03-12 13:24:26.211290316 +0000
+++ Makefile.in	2011-03-12 13:24:37.722688497 +0000
@@ -212,8 +212,8 @@
 #    $Id: Makefile.am 781 2005-12-16 04:57:39Z robert $
 #
 AUTOMAKE_OPTIONS = foreign
-SUBDIRS = include lib osdep examples 
-DIST_SUBDIRS = osdep include lib examples perl python
+SUBDIRS = include lib osdep
+DIST_SUBDIRS = osdep include lib
 pcdir = $(libdir)/pkgconfig
 pc_DATA = libmusicbrainz.pc
 EXTRA_DIST = libmusicbrainz.pc.in README.win32 config_win32.h \
