--- configure.ac.orig	2012-06-04 16:03:32.000000000 +0000
+++ configure.ac	2012-06-04 16:02:39.000000000 +0000
@@ -140,7 +140,7 @@
     case "$optimize" in
 	-O|"-O "*)
 	    optimize="-O"
-	    optimize="$optimize -fforce-mem"
+	    : #x optimize="$optimize -fforce-mem"
 	    optimize="$optimize -fforce-addr"
 	    : #x optimize="$optimize -finline-functions"
 	    : #- optimize="$optimize -fstrength-reduce"
