--- osdep/mb_win32.cpp.old	2006-11-14 16:34:51.000000000 +0000
+++ osdep/mb_win32.cpp	2011-03-12 13:19:56.541289207 +0000
@@ -28,7 +28,7 @@
 #include <windows.h>
 #include "mmsystem.h"
 
-#include "mb_win32.h"           // can't assume symlink mb.h to cdi_win32.h
+#include "mb.h"           // can't assume symlink mb.h to cdi_win32.h
 #include "diskid.h"
 #include "../config_win32.h"    // a copy of a Cygwin generated config.h
 
