--- configure.in.old	2011-03-12 12:59:04.673034093 +0000
+++ configure.in	2011-03-12 12:59:23.891284435 +0000
@@ -120,26 +120,6 @@
  )
 ])
 
-AC_DEFUN([AC_PROTOTYPE_ACCEPT],[
-AC_PROTOTYPE(accept,
- [
-  #include <sys/types.h>
-  #include <sys/socket.h>
- ],
- [
-  int a = 0;
-  ARG2 * b = 0;
-  ARG3 * c = 0;
-  accept(a, b, c);
- ],
- ARG2, [struct sockaddr, void],
- ARG3, [socklen_t, size_t, int, unsigned int, long unsigned int])
-])
-
-AC_DEFINE_UNQUOTED([ACCEPT_ARG2],,"Argument 2 to accept()")
-AC_DEFINE_UNQUOTED([ACCEPT_ARG3],,"Argument 3 to accept()")
-AC_PROTOTYPE_ACCEPT
-
 AC_DEFINE_UNQUOTED(PREFIX, "${prefix}", [Application install prefix])
 
 AC_OUTPUT([
