--- configure.ac-old	2010-07-25 22:14:46.991406820 +0100
+++ configure.ac	2010-07-25 22:16:32.023801444 +0100
@@ -277,7 +277,7 @@
 
   ],
     # program could not be run
-    AC_MSG_RESULT(no)
+    AC_MSG_RESULT(no), AC_MSG_RESULT(yes)
 )
 
 CFLAGS=$OLD_CFLAGS
@@ -661,7 +661,7 @@
 
   ],
     # program could not be run
-    AC_MSG_RESULT(no)
+    AC_MSG_RESULT(no),AC_MSG_RESULT(yes)
 )
 
 CFLAGS=$OLD_CFLAGS
@@ -731,7 +731,7 @@
     AC_DEFINE(HAVE_NEAACDEC_H)
   ],
     # program could not be run
-    AC_MSG_RESULT(no)
+    AC_MSG_RESULT(no),AC_MSG_RESULT(yes)
 )
 
 if test "x$have_faad2" != "xtrue"; then
@@ -765,7 +765,7 @@
 
   ],
     # program could not be run
-    AC_MSG_RESULT(no)
+    AC_MSG_RESULT(no),AC_MSG_RESULT(yes)
 )
 
 fi
