--- configure.in.old	2011-03-12 13:46:26.201283885 +0000
+++ configure.in	2011-03-12 13:53:31.581289516 +0000
@@ -110,7 +110,7 @@
 
   return -1;
 }
-],have_ofaver=true,have_odaver=false,have_ofaver=false)
+],have_ofaver=true,have_odaver=false,have_ofaver=true)
 
 AC_MSG_CHECKING([for version >= 0.4.0 in -lofa])
 if test "$have_ofaver" = "true"; then
@@ -155,7 +155,7 @@
 
   return -1;
 }
-],have_mbver=true,have_mbver=false,have_mbver=false)
+],have_mbver=true,have_mbver=false,have_mbver=true)
 
 AC_MSG_CHECKING([for version >= 2.1.0 in -lmusicbrainz])
 if test "$have_mbver" = "false"; then
@@ -198,7 +198,7 @@
 fi
 
 dnl Check for TagLib 1.4
-AC_CHECK_TAGLIB(1.4, have_taglib=yes, have_taglib=no)
+AC_CHECK_TAGLIB(1.4, have_taglib=yes, have_taglib=yes)
 
 dnl Check for libmpcdec
 AC_CHECK_LIB(mpcdec, mpc_decoder_decode, have_mpcdec=yes,)
