--- lib/fileio.cpp.old	2011-03-12 14:38:27.471909276 +0000
+++ lib/fileio.cpp	2011-03-12 14:36:41.081284628 +0000
@@ -25,6 +25,8 @@
 
 ----------------------------------------------------------------------------*/
 
+#include <process.h>
+#include <io.h>
 #include <assert.h>
 #include <errno.h>
 #include <string>
@@ -106,7 +108,7 @@
 
 int tmkdir(const char *pathname, const char *encoding)
 {
-    return mkdir(utf8ToEncoding(pathname, encoding).c_str(), 0755);
+    return mkdir(utf8ToEncoding(pathname, encoding).c_str());
 }
 
 int trmdir(const char *pathname, const char *encoding)
--- lib/plugins.h.old	2011-03-12 14:32:06.001913983 +0000
+++ lib/plugins.h	2011-03-12 14:39:17.271289250 +0000
@@ -57,11 +57,7 @@
             char            file[MAX_PATH];
             vector<CapInfo> caps;
 
-#ifdef WIN32
-            HMODULE         handle;
-#else
             void           *handle;
-#endif
 };
 
 class Plugins
