--- CMakeLists.txt.orig	2015-02-06 14:40:28.000000000 +0100
+++ CMakeLists.txt	2015-02-06 14:40:39.000000000 +0100
@@ -56,13 +56,11 @@
 set(LIBRARY_OUTPUT_PATH "${CMAKE_BINARY_DIR}/lib" )
 
 # pkg-config
-IF (NOT WIN32)
-  CONFIGURE_FILE (${CMAKE_CURRENT_SOURCE_DIR}/QJson.pc.in
-                  ${CMAKE_CURRENT_BINARY_DIR}/QJson.pc
-                  @ONLY)
-  INSTALL (FILES ${CMAKE_CURRENT_BINARY_DIR}/QJson.pc
-           DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
-ENDIF (NOT WIN32)
+CONFIGURE_FILE (${CMAKE_CURRENT_SOURCE_DIR}/QJson.pc.in
+                ${CMAKE_CURRENT_BINARY_DIR}/QJson.pc
+                @ONLY)
+INSTALL (FILES ${CMAKE_CURRENT_BINARY_DIR}/QJson.pc
+         DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
 
 # Subdirs
 ADD_SUBDIRECTORY(src)
