--- mkspecs/win32-g++/qmake.conf.orig	2012-11-15 20:03:01.000000000 +0000
+++ mkspecs/win32-g++/qmake.conf	2012-11-15 20:05:23.000000000 +0000
@@ -49,10 +49,11 @@
 QMAKE_CXXFLAGS_EXCEPTIONS_ON = -fexceptions -mthreads
 QMAKE_CXXFLAGS_EXCEPTIONS_OFF = -fno-exceptions
 
-QMAKE_INCDIR		=
+QMAKE_INCDIR		= /target/include
+QMAKE_LIBDIR		= /target/bin
 QMAKE_INCDIR_QT		= $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT		= $$[QT_INSTALL_LIBS]
-
+ 
 QMAKE_RUN_CC		= $(CC) -c $(CFLAGS) $(INCPATH) -o $obj $src
 QMAKE_RUN_CC_IMP	= $(CC) -c $(CFLAGS) $(INCPATH) -o $@ $<
 QMAKE_RUN_CXX		= $(CXX) -c $(CXXFLAGS) $(INCPATH) -o $obj $src
