--- src/corelib/codecs/codecs.pri.orig	2012-11-15 14:08:47.000000000 +0000
+++ src/corelib/codecs/codecs.pri	2012-11-15 14:09:22.000000000 +0000
@@ -58,3 +58,26 @@
         }
 }
 symbian:LIBS += -lcharconv
+
+win32 {
+	!contains(QT_CONFIG,iconv) {
+                # no iconv, so we put all plugins in the library
+                HEADERS += \
+                        ../plugins/codecs/cn/qgb18030codec.h \
+                        ../plugins/codecs/jp/qeucjpcodec.h \
+                        ../plugins/codecs/jp/qjiscodec.h \
+                        ../plugins/codecs/jp/qsjiscodec.h \ 
+                        ../plugins/codecs/kr/qeuckrcodec.h \
+                        ../plugins/codecs/tw/qbig5codec.h \
+                        ../plugins/codecs/jp/qfontjpcodec.h
+                SOURCES += \
+                        ../plugins/codecs/cn/qgb18030codec.cpp \
+                        ../plugins/codecs/jp/qjpunicode.cpp \
+                        ../plugins/codecs/jp/qeucjpcodec.cpp \
+                        ../plugins/codecs/jp/qjiscodec.cpp \
+                        ../plugins/codecs/jp/qsjiscodec.cpp \ 
+                        ../plugins/codecs/kr/qeuckrcodec.cpp \
+                        ../plugins/codecs/tw/qbig5codec.cpp \
+                        ../plugins/codecs/jp/qfontjpcodec.cpp
+	}
+}
