--- taglib/toolkit/taglib.h-old	2011-11-27 19:16:08.798370994 +0000
+++ taglib/toolkit/taglib.h	2011-11-27 19:16:14.166470948 +0000
@@ -48,7 +48,6 @@
 #  include <libkern/OSAtomic.h>
 #  define TAGLIB_ATOMIC_MAC
 #elif defined(WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(__CYGWIN__)
-#  define NOMINMAX
 #  include <windows.h>
 #  define TAGLIB_ATOMIC_WIN
 #elif defined (__GNUC__) && (__GNUC__ * 100 + __GNUC_MINOR__ >= 401)    \
